src/connection/database/login_pane.jsx

```diff
@@ -57,7 +61,9 @@ export default class LoginPane extends React.Component {
       );
 
     const captchaPane =
-      l.captcha(lock) && l.captcha(lock).get('required') ? (
+      l.captcha(lock) &&
+      l.captcha(lock).get('required') &&
+      (isHRDDomain(lock, databaseUsernameValue(lock)) || !sso) ? (
```

Comment:
This is the real crux of the change - showing captcha if required, but also for ADFS SSO connections only (other SSO connections should hide the Captcha).